### PR TITLE
Prove `IsUniform.entropy_eq`

### PR DESCRIPTION
**Pull request by Paul-Lez**
I've repurposed this PR to prove `IsUniform.entropy_eq` (originally this had been for the proof of `prob_ge_exp_neg_entropy`, which is now being done in #94)

### CODE REVIEW

**Comment by Paul-Lez** [comment]
There seem to be some builds errors to so I'll fix those too!

File: PFR/Entropy/Basic.lean [hunk 1]
```diff
@@ -306,16 +306,120 @@ lemma entropy_eq_log_card {X : Ω → S} (hX : Measurable X) (μ : Measure Ω) (
...
+
+lemma IsFiniteMeasure_sum_measureReal_univ_inv_mul_measureReal_singleton [Nonempty S] (μ : Measure S)
+  [IsFiniteMeasure μ] [NeZero μ] : ∑ s : S, ((μ.real Set.univ)⁻¹ * μ.real {s}) = 1:= sorry
+
 /-- If $X$ is an $S$-valued random variable, then there exists $s \in S$ such that
 $P[X=s] \geq \exp(-H[X])$. -/
-lemma prob_ge_exp_neg_entropy (X : Ω → S) (μ : Measure Ω) :
-  ∃ s : S, μ.map X {s} ≥ (μ Set.univ) * (rexp (- H[X ; μ])).toNNReal := by sorry
+lemma prob_ge_exp_neg_entropy [Nonempty S] (X : Ω → S) (hX : Measurable X) (μ : Measure Ω)
+  [hμ' : IsFiniteMeasure μ] [NeZero μ] (hbd : ∀ (s : S), μ.map X {s} ≠ 0) :
```
Comment (llllvvuu) [inline PFR/Entropy/Basic.lean:344]: are we allowed to add these assumptions?
- `Nonempty S` is already added via `variables` in this file
- we can derive `hμ'` and `NeZero \mu` via `rcases` (the statement degenerates if either is false)
- not sure about `hX` and `hbd`. IIRC i didn't need `hX`.

Comment (llllvvuu) [reply PFR/Entropy/Basic.lean:344]: What I ended up doing for `hbd` is to use `Finset.filter` to specify only summing nonzero values; not sure if it was the most efficient way though

Comment (Paul-Lez) [reply PFR/Entropy/Basic.lean:344]: A lot of the assumptions can and should be removed (I added them as I went as was planning to remove them in the golfing process :)). 
 For `hbd`, I guess in my proof too I could be removed by writing H as a sum over the elements with non zero probability (maybe this fact should be stated in a separate lemmas?)

**Pull request merged**